sky/data/storage.py

```diff
@@ -1871,6 +1904,540 @@ def _delete_gcs_bucket(self, bucket_name: str) -> bool:
+                        self.resource_group_name, {'location': self.region})
+            # check if the storage account name already exists under the given
+            # resource group name.
+            try:
+                self.storage_client.storage_accounts.get_properties(
+                    self.resource_group_name, self.storage_account_name)
+            except azure.exceptions().ResourceNotFoundError as e:
+                if 'Code: ResourceNotFound' in e.message:
+                    try:
+                        creation_response = self.storage_client.storage_accounts.begin_create(
+                            self.resource_group_name, self.storage_account_name, {
+                                'sku': {
+                                    'name': 'Standard_GRS'
+                                },
+                                'kind': 'StorageV2',
+                                'location': self.region,
+                                'encryption': {
+                                    'services': {
+                                        'blob': {
+                                            'key_type': 'Account',
+                                            'enabled': True
+                                        }
+                                    },
+                                    'key_source': 'Microsoft.Storage'
+                                },
+                            }).result()
+                        # Assigning Storage Blob Data Owner role of the storage account
+                        # Reference: https://github.com/Azure/azure-sdk-for-python/issues/35573 # pylint: disable=line-too-long
+                        authorization_client = data_utils.create_az_client(
+                            'authorization')
+                        graph_client = data_utils.create_az_client('graph')
+                        async def get_object_id():
+                            import logging
+                            original_level = logging.getLogger("httpx").getEffectiveLevel()
+                            # silencing the INFO level response log from httpx request
+                            logging.getLogger("httpx").setLevel(logging.WARNING)
+                            user = await graph_client.users.with_url(
+                                'https://graph.microsoft.com/v1.0/me').get()
+                            logging.getLogger("httpx").setLevel(original_level)
+                            object_id = str(user.additional_data['id'])
+                            return object_id
+                        object_id = asyncio.run(get_object_id())
+                        # Defintion ID of Storage Blob Data Owner role.
+                        # Reference: https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/storage#storage-blob-data-owner # pylint: disable=line-too-long
+                        storage_blob_data_owner_role_id = 'b7e6dc6d-f1e8-4753-8033-0f276bb0955b'
```

Comment:
Do you mind elaborating?

Reply:
i mean have a constant for `storage_blob_data_owner_role_id`, i.e. `AZ_BLOB_DATA_OWNER_ROLE_ID = 'b7e6dc6d-f1e8-4753-8033-0f276bb0955b'`, instead of scattering all of those (seemingly) random raw strings inside the code

Reply:
similar to `'ERROR: AADSTS50020'`

Reply:
Ah, I see. For `storage_blob_data_owner_role_id = 'b7e6dc6d-f1e8-4753-8033-0f276bb0955b'`, it's a fixed value that's only used once in this function when creating the storage account. It's not used any where else. Do you still think we need to make it globally available constant? Same question to `'ERROR: AADSTS50020'`

Reply:
That might also works